Sources/MapLibreSwiftUI/MapViewCoordinator.swift

```diff
@@ -0,0 +1,194 @@
+import Foundation
+import MapLibre
+import MapLibreSwiftDSL
+
+public class MapViewCoordinator: NSObject {
+    
+    // This must be weak, the UIViewRepresentable owns the MLNMapView.
+    weak var mapView: MLNMapView?
+    var parent:  MapView
+
+    // Storage of variables as they were previously; these are snapshot
+    // every update cycle so we can avoid unnecessary updates
+    private var snapshotUserLayers: [StyleLayerDefinition] = []
+    private var snapshotCamera: MapViewCamera?
+    private var onGestureEnd: (MLNMapView, UIGestureRecognizer) -> Void
+    
+    init(parent: MapView,
+         onGestureEnd: @escaping (MLNMapView, UIGestureRecognizer) -> Void) {
+        self.parent = parent
+        self.onGestureEnd = onGestureEnd
+    }
+    
+    // MARK: Core UIView Functionality
+    
+    @objc func captureGesture(_ sender: UIGestureRecognizer) {
+        guard let mapView, sender.state == .ended else {
+            return
+        }
+        
+        onGestureEnd(mapView, sender)
+    }
+
+    // MARK: - Coordinator API - Camera + Manipulation
+
+    func updateCamera(mapView: MLNMapView, camera: MapViewCamera, animated: Bool) {
+        guard camera != snapshotCamera else {
+            // No action - camera has not changed.
+            return
+        }
+        
+        switch camera.state {
+        case .centered:
+            mapView.userTrackingMode = .none
+            mapView.setCenter(camera.coordinate,
+                              zoomLevel: camera.zoom,
+                              direction: camera.course,
+                              animated: animated)
+        case .trackingUserLocation:
+            mapView.userTrackingMode = .follow
+            mapView.setZoomLevel(camera.zoom, animated: false)
+        case .trackingUserLocationWithHeading:
+            mapView.userTrackingMode = .followWithHeading
+            mapView.setZoomLevel(camera.zoom, animated: false)
+        case .trackingUserLocationWithCourse:
+            mapView.userTrackingMode = .followWithCourse
+            mapView.setZoomLevel(camera.zoom, animated: false)
+        case .rect, .showcase:
+            // TODO: Need a method these/or to finalize a goal here.
+            break
+        }
+        
+        if let pitch = camera.pitch {
+            mapView.minimumPitch = pitch
+            mapView.maximumPitch = pitch
+        } else {
+            mapView.minimumPitch = 0
+            mapView.maximumPitch = 90
```

Comment:
90 feels pretty steep! Are you sure you didn't mean 60? That's the classical maximum.

Reply:
To double check: thought this was "straight on" but Ian thinks that this is actually zero. Verify.